packages/node-sdk/sm/primitives/evm/rpc/paima-l2.ts

```
@@ -199,6 +206,25 @@ export default function* processPaimaL2SyncProtocolResponse(
       const { parsed } = batchedMessage;
       const { userAddress, millisecondTimestamp, userSignature, gameInput } =
         parsed;
+      // 24h timestamp validation for batched inputs
+      if (blockTimestampMs !== undefined) {
+        const signedTs = Number(millisecondTimestamp);
+        if (
+          !Number.isFinite(signedTs) ||
```

if you want we can throw an error if `!Number.isFinite(signedTs)`